### PR TITLE
Add Traditional Chinese, Hong Kong translation (zh-HK)

### LINKED ISSUE #1524: Add Traditional Chinese, Hong Kong translation (zh-HK)
Hello. For zh-HK users, Mastodon is displaying Simplified Chinese interface. This is not good.

Need to improve the situation.

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by yookoala**
Fix #1524

### CODE REVIEW

**Comment by yookoala** [comment]
Need help. How do I fix the codeclimate report? Or do I need to do it?

**Comment by yookoala** [comment]
I think the "similar code report" is entirely invalid. Those are completely un-related code.
It is also not related to the code in my commits.

I have no permission to dismiss the report. Please help.

File: app/assets/javascripts/components/locales/zh-hk.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  "column.notifications": "通知",
+  "column.public": "跨站公共時間軸",
+  "compose_form.placeholder": "你在想甚麼？",
+  "compose_form.privacy_disclaimer": "你的私人文章，將被遞送至你所提及的 {domains} 用戶。你是否信任 {domainsCount, plural, one {這個網站} other {這些網站}}？請留意，文章私隱設定只適用於各 Mastodon 服務站，如果 {domains} {domainsCount, plural, one {不是 Mastodon 服務站} other {之中有些不是 Mastodon 服務站}}，對方將無法收到這篇文章的私隱設定，然後可能被轉推給不能預知的用戶閱讀。",
+  "compose_form.private": "標示為「只有關注你的人能看」",
+  "compose_form.publish": "發文",
+  "compose_form.sensitive": "將媒體檔案標示為「敏感內容」",
+  "compose_form.spoiler": "將部份文字藏於警告訊息之後",
+  "compose_form.unlisted": "請勿在公共時間軸顯示",
+  "empty_column.community": "本站時間軸暫時未有內容，快公開寫些甚麼來佔首發啊！",
```
Comment (tommyku) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: I think "快貼文來搶頭香啊！" sounds more Hong Kong than "快公開寫些甚麼來佔首發啊！"

File: app/assets/javascripts/components/locales/zh-hk.jsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  "report.target": "Reporting",
+  "search.account": "用戶",
+  "search.hashtag": "標籤",
+  "search.placeholder": "搜尋",
+  "search_results.total": "{count} 項結果",
+  "search.status_by": "按用戶名稱搜尋文章",
+  "status.delete": "刪除",
+  "status.favourite": "喜歡",
+  "status.load_more": "載入更多",
+  "status.media_hidden": "隱藏起媒體內容",
```
Comment (tommyku) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: possibly redundant "起"

File: config/locales/devise.zh-HK.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      no_token: 你必須使用重設密碼電郵內的網址進入本頁。如果你確是使用電郵內的網址，請確認你用了完整的網址。
+      send_instructions: 你將在幾分鐘內收到重設密碼的電郵指示。
+      send_paranoid_instructions: 如果你的電郵地址已經存在於我們的資料庫，你將會在幾分鐘內收到重設密碼的電郵指示。
+      updated: 你的密碼已經更新，你現在正登入本站。
+      updated_not_active: 你的密碼已經更新。
+    registrations:
+      destroyed: 再見了！你的用戶已被取消，希望我們相有相見的機會吧。
+      signed_up: 歡迎你！你的登記已經成功。
+      signed_up_but_inactive: 你的登記已經成功，可是由於你的用戶還被被啟用，暫時還不能讓你登入。
+      signed_up_but_locked: 你的登記已經成功，可是由於你的戶已被鎖定，我們無法讓你登入。
```
Comment (tommyku) [inline config/locales/devise.zh-HK.yml]: missing "用" for "用戶"

**Comment by yookoala** [comment]
Text modified according to comments.

**Comment by yookoala** [comment]
Has encountered "notifications_controller_spec.rb" error before in my Travis. It happens occasionally. Just trigger a rebuild could got that "fix". I'm sure the tests can be improved, but that is not the point of this commit.

**Comment by yookoala** [comment]
Rebased commits on the current master.

**Review by wxcafe — APPROVED** [review]

File: app/assets/javascripts/components/locales/zh-hk.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  "account.follows_you": "關注你",
+  "account.follows": "正在關注",
+  "account.mention": "提及 @{name}",
+  "account.posts": "文章",
+  "account.requested": "等候審批",
+  "account.unblock": "解除對 @{name} 的封鎖",
+  "account.unfollow": "取消關注",
+  "column_back_button.label": "先前顯示",
+  "column.community": "本站時間軸",
+  "column.home": "家",
```
Comment (siuying) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: Home -> 主頁?

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 界面上的 "Home" 只是一個 Column，所以沒有用「主頁」而用「家」（想不到可以另外怎樣譯）

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 類似道理，右欄上的 "Back" 更難譯，根本只是讓右欄回到上一個 view 而已，所以譯「先前顯示」

**Comment by yookoala** [comment]
Notes about the translation of the key terms (Google Drive):
https://goo.gl/3uZcjh

File: app/assets/javascripts/components/locales/zh-hk.jsx (hunk 4 above)
Comment (chaofan) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 一般是「回退」比较常见。

Comment (tommyku) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 或「返回」

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 「返回」的確比較簡潔

讓我開一個新 feature branch 做修改吧，但會等一批 comment 再一次過 PR

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: Home 也許可以譯作「主介面」或者「主時間軸」？

Comment (tommyku) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 「主頁」？

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 問題係除非你用手機睇，否則佢個「Home」唔係「頁」。見下圖︰

Comment (chaofan) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: mac 的译法
image

Comment (siuying) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 至少 Twitter 是把它們的 Home 稱為主頁，那不是一頁而是一個名稱
screen shot 2017-04-14 at 11 06 08 am

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: 還是覺得怪怪的……

但有先例的話，就先改「主頁」吧

Comment (yookoala) [inline app/assets/javascripts/components/locales/zh-hk.jsx]: See: yookoala/mastodon@641976b

**Pull request merged**